view/frontend/web/js/view/payment/method-renderer/vault.js

```diff
@@ -11,50 +11,244 @@
+
     return VaultComponent.extend({
         defaults: {
-            template: 'Magento_Vault/payment/form'
+            template: 'Adyen_Payment/payment/card-vault-form.html',
+            checkoutComponentBuilt: false,
+            modalLabel: 'card_action_modal'
+        },
+
+        initObservable: function () {
+            this._super()
+                .observe([
+                    'checkoutComponentBuilt'
+                ]);
+
+            return this;
+        },
+
+        initialize: function () {
+            let self = this;
+            this._super();
+            let paymentMethodsObserver = adyenPaymentService.getPaymentMethods();
+            paymentMethodsObserver.subscribe(
+                function (paymentMethodsResponse) {
+                    self.loadCheckoutComponent(paymentMethodsResponse)
+                });
+            self.loadCheckoutComponent(paymentMethodsObserver());
+
+            return this;
+        },
+
+        loadCheckoutComponent: async function(paymentMethodsResponse) {
+            this.checkoutComponent = await adyenCheckout.buildCheckoutComponent(
+                paymentMethodsResponse,
+                this.handleOnAdditionalDetails.bind(this)
+            );
+
+            if (this.checkoutComponent) {
+                this.checkoutComponentBuilt(true)
+            }
+        },
+
+        // TODO: implement handleOnAdditionalDetails
+        handleOnAdditionalDetails: function (result) {
+            var self = this;
+            var request = result.data;
+            request.orderId = self.orderId;
+            console.log('Hello in handleOnAdditionalDetails');
+        },
+
+        renderCardVaultToken: function () {
+            let self = this;
+            if (!this.getClientKey()) {
+                return false
+            }
+
+            let componentConfig = {
+                hideCVC: false,
+                brand: this.getCardType(),
+                storedPaymentMethodId: this.getGatewayToken(),
+                expiryMonth: this.getExpirationMonth(),
+                expiryYear: this.getExpirationYear(),
+                //holderName: 'First tester',
+                onChange: this.handleOnChange.bind(this)
+            }
+
+            self.component = adyenCheckout.mountPaymentMethodComponent(
+                this.checkoutComponent,
+                'card',
+                componentConfig,
+                '#cvcContainer-' + this.getId()
+            )
+            this.component = self.component
+
+            return true
+        },
+
+        handleOnChange: function (state, component) {
+            validTokens[this.getId()] = !!state.isValid;
+            isValidObserver(validTokens)
+        },
+
+        isButtonActive: function () {
+            return this.isActive() && this.isPlaceOrderActionAllowed() && isValidObserver()[this.getId()];
+        },
+
+        // TODO check: installments, add modal and handleOnAdditionalDetails for 3ds2
```

Comment:
We have another TODO related to the action modal above, shall we delete this one? Also, we can create a separate ticket to investigate the installments for recurring payments.

Reply:
Yea we can do this